src/components/AddItem.js

```diff
@@ -1,34 +1,78 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import { withFirestore } from 'react-firestore';
+import '../styles/AddItem.css';
+import { v4 as uuidv4 } from 'uuid';
 
 const AddItem = props => {
+  const emptyShoppingItem = {
+    name: '',
+    lastPurchasedDate: null,
+    nextPurchase: 7,
+  };
+
   const { firestore } = props;
-  const [enteredValue, setEnteredValue] = useState();
+  const [enteredValue, setEnteredValue] = useState(emptyShoppingItem);
 
-  const addMsg = e => {
+  const handleChange = e => {
+    setEnteredValue({ ...enteredValue, [e.target.name]: e.target.value });
+  };
+
+  const addItem = e => {
     e.preventDefault();
-    firestore.collection('example').add({
-      msg: enteredValue,
-    });
 
-    setEnteredValue('');
+    if (enteredValue.name === '') {
+      alert('Please enter an item name');
+    } else {
+      firestore.collection('shoppingList').add({
+        name: enteredValue.name,
+        nextPurchase: parseInt(enteredValue.nextPurchase, 10),
+        token: localStorage.getItem('userToken'),
+      });
+
+      resetInput();
+    }
   };
+
+  const resetInput = () => {
+    setEnteredValue(emptyShoppingItem);
+  };
+
+  useEffect(() => {
+    const userToken = uuidv4();
+    localStorage.setItem('userToken', userToken);
+  }, []);
```

Comment:
I know for testing purposes token was required here. I think we can remove the useEffect here since `Welcome.js` component already generates the token for us.

Reply:
Yep! This is perfect for testing, but before we merge this into master, let's get the Welcome Prompt PR merged in and rebase this with master, remove this and replace it with the new token logic, test it again, and merge it! Happy to help with all the rebasing here if necessary! ❤️